L13-Python-Interceptors.md

```diff
@@ -0,0 +1,285 @@
+Application Gets the Response
+```
+
+Also, this implementation empowers developers to invoke the `Continuation` zero or multiple time if necessary, which is useful in case of authentication or retry.
+
+### Server-Side API
+
+On the server side, there is also an RPC description class, which only contains two property, `method` and `invocation_metadata`. Unlike the client side, the `method` variable is read-only.
+
+```Python
+class HandlerCallDetails(six.with_metaclass(abc.ABCMeta)):
+    """Describes an RPC that has just arrived for service.
+
+    Attributes:
+      method: The method name of the RPC.
+      invocation_metadata: The metadata sent by the client.
+    """
+```
+
+The `Continuation` in the server side is thinner than the client side. It is a function that takes a `grpc.HandlerCallDetails` and proceeds to invoke the next interceptor in the chain, if any, or the RPC handler lookup logic, with the call details, passed as an argument and returns a `grpc.RpcMethodHandler` instance if the RPC is considered serviced, or None otherwise.
+
+The server-side interceptor interface contains one abstract member function that takes in `Continuation`, and `grpc.HandlerCallDetails`, then returns a `grpc.RpcMethodHandler`.
+
+```Python
+class ServerInterceptor(six.with_metaclass(abc.ABCMeta)):
+    @abc.abstractmethod
+    def intercept_service(self, continuation, handler_call_details): pass
+```
+
+To apply interception, the [server construction function](https://grpc.io/grpc/python/grpc.html#grpc.server) takes in an `interceptors` parameter that accepts optional list of `grpc.ServerInterceptor` instance. The usage snippet will look like:
+
+```Python
+class ExampleInterceptor(grpc.ServerInterceptor):
+    def intercept_service(self, continuation, handler_call_details):
+        ...
+
+server_interceptor_1 = ExampleInterceptor(...)
+server = grpc.server(..., interceptors=[server_interceptor_1])
+```
+
+### Server-Side Implementation
+
+In the implementation, there is a new concept named "Service Pipeline". It assembles the server side interceptors and is responsible for generating the final `grpc.RpcMethodHandler` to be invoked for incoming requests. The generation process will happen every time the server receives an RPC, the execution flow will look like:
+
+```
+Server Receives a Request ->
+    Interceptor A Start ->
+        Interceptor B Start ->
+            Interceptor C Start ->
+                The Original Handler
+            Interceptor C Returns Updated Handler C ->
+        Interceptor B Returns Updated Handler B ->
+    Interceptor A Returns Updated Handler A ->
+
+    Invoke the Updated Handler A with the Request ->
+    Updated Handler A Returns Response ->
+Server Replies
+```
+
+The most significant difference with client-side is that the interception happens ahead of the actual invocation of application behavior. So the interceptor function will able to access neither the request nor servicer context.
```

Comment:
"with client-side" -> "on the client side"

Reply:
Updated to "difference compared to the client side."